spec/graphql_connector/query_builder_spec.rb

```diff
@@ -4,16 +4,60 @@
 
 module GraphqlConnector
   describe QueryBuilder do
-    let(:builder) { described_class.new(model, conditions, selected_fields) }
+    let(:builder) do
+      described_class.new(type, model, inputs, selected_fields)
+    end
+    let(:type) { 'query' }
     let(:model) { 'product' }
-    let(:conditions) { { id: 1 } }
+    let(:inputs) { { id: 1 } }
     let(:selected_fields) { %w[id name] }
     subject { builder.create }
 
-    let(:expect_result) { 'query { product(id: 1) { id name } }' }
+    let(:expect_result) do
+      <<-STRING
+      query {
+        product(id: 1) {
+          id name
+        }
+      }
+      STRING
+    end
+
+    it { is_expected.to eq(expect_result) }
+
+    context 'with mutation type' do
+      let(:type) { 'mutation' }
+
+      let(:expect_result) do
+        <<-STRING
+      mutation {
+        product(id: 1) {
+          id name
+        }
+      }
+        STRING
+      end
+
+      it { is_expected.to eq(expect_result) }
+
+      context 'with nested input attributes format' do
+        let(:inputs) do
+          { input: { attributes: { id: 1 } } }
+        end
+        let(:selected_fields) { [product: %i[id name]] }
+
+        let(:expect_result) do
+          <<-STRING
+      mutation {
+        product(input: {attributes: {id: 1}}) {
+          product { id name }
+        }
+      }
+          STRING
+        end
 
-    it 'returns a valid graphql query string' do
-      subject.equal? expect_result
```

Comment:
@Garllon 

Refactored this spec since as of [`equal?`](https://apidock.com/ruby/Object/equal%3F) comparing whether it would always return a String object. Or was this intended (like `subject.is_a?(String)`)?

Reply:
Further updated code and splitted `query_builder` into smaller templates --> See also [here](https://github.com/Garllon/graphql_connector/pull/21/commits/320c93a11d059ce7148429d9befda0f137999d19)